### PR TITLE
[luci] Support CONCAT in ConvertNCHWToNHWCPass

### PR DESCRIPTION
**Pull request by jinevening**
This supports CONCAT in CovnertNCHWToNHWCPass.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/5845

### CODE REVIEW

File: compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp [hunk 1]
```diff
@@ -87,6 +87,21 @@ luci::CircleTranspose *create_4d_transpose(luci::CircleNode *node,
...
+  if (pos_axis == 0)
+    return 0;
+  else if (pos_axis == 1)
+    return 3;
+  else if (pos_axis == 2)
+    return 1;
+  else if (pos_axis == 3)
+    return 2;
+  else
+    throw std::runtime_error("Concat axis must be in range [-4, 4)");
```
Comment (seanshpark) [inline compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp:102]: How about something like this?
```cxx
uint32_t pos_axis = axis >= 0 ? static_cast<uint32_t>(axis) : static_cast<uint32_t>(axis + 4);
static const uint32_t to_nhwc[4] = {0, 3, 1, 2};
if (pos_axis > 3)
  throw std::runtime_error("Concat axis must be in range [-4, 4)");
return to_nhwc[pos_axis];
```

File: compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp [hunk 2]
```diff
@@ -444,6 +459,32 @@ class ConvertNCHWToNHWC final : public luci::CircleNodeMutableVisitor<bool>
     return true;
   }
 
+  bool visit(luci::CircleConcatenation *node)
+  {
+    const auto num_values = node->numValues();
```
Comment (seanshpark) [inline compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp:464]: I think we need to check rank of node and if not 4 then return falsel;

Comment (jinevening) [reply compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp:464]: node rank is checked before the visitor is called.

Related code
```cpp
...
     ConvertNCHWToNHWC converter;
      auto circle_node = loco::must_cast<luci::CircleNode *>(node);
      if (circle_node->rank() != 4)
        continue;

      if (circle_node->accept(&converter))
      {
        set_data_format(node, DataFormat::NHWC);
        changed = true;
      }
...
```

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**